Ninja fails due to type casting errors

When I try `ninja -C build test` I get the following:
```
(msprime-dev) nbaya@x86_64-apple-darwin13 lib % ninja -C build test
ninja: Entering directory `build'
[2/15] Compiling C object libmsprime.a.p/mutgen.c.o
FAILED: libmsprime.a.p/mutgen.c.o
x86_64-apple-darwin13.4.0-clang -Ilibmsprime.a.p -I. -I.. -Isubprojects/kastore -I../subprojects/kastore -Isubprojects/tskit -I../subprojects/tskit -I../subprojects/tskit/tskit -I/usr/local/Cellar/gsl/2.7/include -Xclang -fcolor-diagnostics -pipe -Wall -Winvalid-pch -g -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -isystem /Users/nbaya/anaconda3/envs/msprime-dev/include -D_FORTIFY_SOURCE=2 -mmacosx-version-min=10.9 -isystem /Users/nbaya/anaconda3/envs/msprime-dev/include -std=c99 -Wall -Wextra -Werror -Wpedantic -W -Wmissing-prototypes -Wstrict-prototypes -Wconversion -Wshadow -Wpointer-arith -Wcast-align -Wcast-qual -Wwrite-strings -Wnested-externs -fshort-enums -fno-common -MD -MQ libmsprime.a.p/mutgen.c.o -MF libmsprime.a.p/mutgen.c.o.d -o libmsprime.a.p/mutgen.c.o -c ../mutgen.c
../mutgen.c:299:24: error: cast from 'char *' to 'int32_t *' (aka 'int *') increases required alignment from 1 to 4 [-Werror,-Wcast-align]
    mutation_type_id = (int32_t *) (dest + n);
                       ^~~~~~~~~~~~~~~~~~~~~~
../mutgen.c:302:23: error: cast from 'char *' to 'float *' increases required alignment from 1 to 4 [-Werror,-Wcast-align]
    selection_coeff = (float *) (dest + n);
                      ^~~~~~~~~~~~~~~~~~~~
../mutgen.c:305:20: error: cast from 'char *' to 'int32_t *' (aka 'int *') increases required alignment from 1 to 4 [-Werror,-Wcast-align]
    subpop_index = (int32_t *) (dest + n);
                   ^~~~~~~~~~~~~~~~~~~~~~
../mutgen.c:308:25: error: cast from 'char *' to 'int32_t *' (aka 'int *') increases required alignment from 1 to 4 [-Werror,-Wcast-align]
    origin_generation = (int32_t *) (dest + n);
                        ^~~~~~~~~~~~~~~~~~~~~~
4 errors generated.
[4/15] Compiling C object dev-cli.p/dev-tools_argtable3.c.o
ninja: build stopped: subcommand failed.
```
Here are my compiler versions
```
(msprime-dev) nbaya@x86_64-apple-darwin13 lib % cc --version
Apple clang version 11.0.3 (clang-1103.0.32.62)
Target: x86_64-apple-darwin19.6.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin

(msprime-dev) nbaya@x86_64-apple-darwin13 lib % gcc --version
Configured with: --prefix=/Library/Developer/CommandLineTools/usr --with-gxx-include-dir=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/4.2.1
Apple clang version 11.0.3 (clang-1103.0.32.62)
Target: x86_64-apple-darwin19.6.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin

(msprime-dev) nbaya@x86_64-apple-darwin13 lib % clang --version
clang version 10.0.0
Target: x86_64-apple-darwin19.6.0
Thread model: posix
InstalledDir: /Users/nbaya/anaconda3/envs/msprime-dev/bin
```

--- a/lib/mutgen.c
+++ b/lib/mutgen.c
@@ -288,28 +288,22 @@ mutation_matrix_free(mutation_model_t *self)
 static void
 copy_slim_mutation_metadata(slim_mutator_t *params, char *dest, double time)
 {
-    size_t n;
-    int32_t *mutation_type_id;
-    float *selection_coeff;
-    int32_t *subpop_index;
-    int32_t *origin_generation;
-    int8_t *nucleotide;
+    int32_t mutation_type_id = params->mutation_type_id;
+    float selection_coeff = 0;
+    int32_t subpop_index = TSK_NULL;
+    int32_t origin_generation = params->slim_generation - (int32_t) time;
+    int8_t nucleotide = -1;
 
-    n = 0;
-    mutation_type_id = (int32_t *) (dest + n);
-    *mutation_type_id = params->mutation_type_id;
-    n += sizeof(int32_t);
-    selection_coeff = (float *) (dest + n);
-    *selection_coeff = 0.0;
-    n += sizeof(float);
-    subpop_index = (int32_t *) (dest + n);
-    *subpop_index = TSK_NULL;
-    n += sizeof(int32_t);
-    origin_generation = (int32_t *) (dest + n);
-    *origin_generation = params->slim_generation - (int32_t) time;
-    n += sizeof(int32_t);
-    nucleotide = (int8_t *) (dest + n);
-    *nucleotide = -1;
+    memcpy(dest, &mutation_type_id, sizeof(mutation_type_id));
+    dest += sizeof(mutation_type_id);
+    memcpy(dest, &selection_coeff, sizeof(selection_coeff));
+    dest += sizeof(selection_coeff);
+    memcpy(dest, &subpop_index, sizeof(subpop_index));
+    dest += sizeof(subpop_index);
+    memcpy(dest, &origin_generation, sizeof(origin_generation));
+    dest += sizeof(origin_generation);
+    memcpy(dest, &nucleotide, sizeof(nucleotide));
+    dest += sizeof(nucleotide);
 }
 
 static void

